### PR TITLE
Comments now fully working

### PR DESCRIPTION
**Pull request by linster**
Merge with caution. This one's gonna be dangerous.

### CODE REVIEW

File: QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/AnswerViewActivity.java [hunk 1]
```diff
@@ -54,10 +58,11 @@ protected void onCreate(Bundle savedInstanceState) {
 		((TextView) findViewById(R.id.answer_username)).setText(this.answer.getAuthor());
 		
 		/* Populate comment list */
-		List<Comment<Answer>> cl = new ArrayList<Comment<Answer>>();
+		//created as class variable.
 		cl.addAll(answer.getCommentList());
```
Comment (stephenjust) [inline QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/AnswerViewActivity.java:62]: cl = answer.getCommentList(); is sufficient...

File: QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/AnswerViewActivity.java [hunk 2]
```diff
@@ -54,10 +58,11 @@ protected void onCreate(Bundle savedInstanceState) {
...
 			//Set list adapter
 		final CommentListAdapter<Answer> cla = new CommentListAdapter<Answer>(this, R.layout.list_comment, cl);
+		ucla = cla;
```
Comment (stephenjust) [inline QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/AnswerViewActivity.java:65]: Not UC Berkeley?

File: QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/view/ViewCommentDialogFragment.java [hunk 3]
```diff
@@ -25,20 +25,27 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
...
 		
-		UUID questionId = UUID.fromString(getArguments().getString("questionId"));
-		UUID answerId = UUID.fromString(getArguments().getString("answerId"));
-		UUID commentId = UUID.fromString(getArguments().getString("commentId"));
+		//UUID.fromString NPEs if property is missing. Need to fix this.
+		UUID questionId = null;
+		UUID answerId = null;
+		UUID commentId = null;
+		
+		//These two should never be null/empty.
```
Comment (stephenjust) [inline QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/view/ViewCommentDialogFragment.java:33]: You could just check (if .. != null) { ... }

**Comment by stephenjust** [comment]
Also pull from upstream and resolve merge conflicts so travis can run :+1:

File: QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/view/ViewCommentDialogFragment.java (hunk 3 above)
Comment (linster) [inline QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/view/ViewCommentDialogFragment.java:33]: No. Uuid.from string actually throws an npe.

**Pull request merged**